### DISCUSSION TITLE
Xcode already installed but still getting "Missing Xcode, please install"

### DISCUSSION THREAD

**Discussion by meehjeevan** [axmolengine/axmol#1980] [Q&A]
I'm using:
macbook m3 pro
macOS: 14.3
Xcode: 15.4
Axmol: latest from git

I was trying to build for ios simulator with this command
`axmol build -p ios -a arm64 -sdk simulator -c
`

But I got these errors though I have xcode installed:
```
1kiss: info: the cmd follow symlink /opt/homebrew/bin/cmake ==> /opt/homebrew/Cellar/cmake/3.29.5/bin/cmake
1kiss: Using cmake: /opt/homebrew/bin/cmake, version: 3.29.5
1kiss: Building target ios on macos with toolchain clang ...
Exception: /Users/userxx/Documents/tools/axmol/1k/1kiss.ps1:1246
Line |
1246 |          throw "Missing Xcode, please install"
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Missing Xcode, please install
```

**Comment by rh101** [comment]
Have you installed the Xcode command line tools?  If not, try `xcode-select --install`

**Reply by meehjeevan** [reply]
yes command line tools are also installed.

**Reply by rh101** [reply]
What about accepting the Xcode agreement? `sudo xcodebuild -license accept`

Otherwise, open up the `1kiss.ps1` script, and check what the condition is that would be throwing the error.

**Reply by meehjeevan** [reply]
yes, first I switched the developer folder
`sudo xcode-select -switch /Applications/Xcode.app/Contents/Developer` 
and then
`sudo xcodebuild -license accept`
the build worked but now I'm getting this error while trying to run the project in the simulator, I built for devices and also tried to run on the connected device the same errors occurred.
<img width="958" alt="Screenshot 2024-06-10 at 22 26 29" src="https://github.com/axmolengine/axmol/assets/6400139/a97047cf-a73a-4b27-9993-c27c50460b19">
<img width="664" alt="Screenshot 2024-06-10 at 22 27 40" src="https://github.com/axmolengine/axmol/assets/6400139/f380630b-ace9-4fc2-9191-260c0f70d86b">

**Reply by smilediver** [reply]
`Config.h` error seems like Axmol issue. The header is probably included in .c or .m file so it's being compiled as C or Objective-C. The `namespace` stuff should be wrapped in `#ifdef __cplusplus`.

Also this `else` part doesn't make sense and probably should be removed:
```
#ifdef __cplusplus
...
#else
#    define NS_AX_BEGIN
#    define NS_AX_END
#    define USING_NS_AX
#    define NS_AX
#endif
```
If code has these defines, then it contains C++ specific code and wouldn't compile for other languages anyway.

**Reply by rh101** [reply]
> the build worked but now I'm getting this error while trying to run the project in the simulator, I built for devices and also tried to run on the connected device the same errors occurred.

Is this a build of a new project without any modifications?  

I've just tested this using the latest axmol dev branch (on a Mac M1):

1. Re-ran `./setup.ps1` to ensure Axmol is set up correctly
2. Created a new project with `axmol new`
3. Created both a simulator project via `axmol build -p ios -a arm64 -sdk simulator -c`, and a device project via `axmol build -p ios -a arm64 -c`
4. Built both projects in Xcode 15.4 successfully

**Reply by meehjeevan** [reply]
new blank project runs well but a project migrated to axmol which is running well on android is giving such errors on ios.

**Reply by meehjeevan** [reply]
I cleaned and rebuild several times and now I'm getting this LaunchScreen error:
`proj.ios_mac/ios/targets/ios/LaunchScreen.storyboard: Failed to find or create execution context for description '<IBCocoaTouchPlatformToolDescription: 0x600002573100> System content for IBVisionIdiom-SevenAndLater <IBScaleFactorDeviceTypeDescription: 0x600002572900> scaleFactor=2x, renderMode.identifier=(null)'.
`

**Reply by halx99** [reply]
Try execute `xcodebuild -version` , show your output

**Reply by meehjeevan** [reply]
Xcode 15.4
Build version 15F31d

**Reply by smilediver** [reply]
https://stackoverflow.com/questions/76535257/xcode-15-beta-2-build-error-failed-to-find-or-create-execution-context-for-d

I had to up the deployment target.

**Reply by meehjeevan** [reply]
it's really strange.. sometimes it throws error in config.h and hlookup.h like above in the screenshot and sometimes in launchScreen.

**Reply by halx99** [reply]
> Xcode 15.4 Build version 15F31d

Strange, my mac also Xcode 15.4 and works well

**Reply by smilediver** [reply]
> it's really strange.. sometimes it throws error in config.h and hlookup.h like above in the screenshot and sometimes in launchScreen.

It just depends which files Xcode was building first, or which one it decided to show. All the errors are still there.

**Comment by thienphuoc** [comment]
Ensure that xcode-select is pointing to the correct Xcode installation. You can set it using:

```
sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
```

or

```
sudo xcode-select -s /Applications/Xcode.app/Contents/Developer
```

Then you run the axmol build command as usual.

**Reply by meehjeevan** [reply]
yes I did that already
